docs/source/future/core/nrunner.rst

```diff
@@ -151,34 +151,68 @@ create Runnables based on the user's input from the command line::
+will be employed when selecting a Runner.
 
-  runnable_instance = create_runnable('exec', uri='/bin/sleep')
-  if runnable_instance.kind == 'exec':
-     runner = create_runner_exec(runnable_instance)
+A Runner can take different forms, depending on which layer one is
+interacting with.  At the lowest layer, a Runner may be a Python class
+that inherits from :class:`avocado.core.nrunner.BaseRunner`, and
+implements at least a matching constructor method, and a ``run()``
+method that should yield dictionary(ies) as result(s).
 
-A runner should produce status information on the progress of the
-execution of a runnable.  If the runnable produces interesting
-information, it should forward that along.  For instance, using the
-``exec`` runner example, it's helpful to start producing status
-that the process has been created and it's running as soon as
-possible.  These can be as simple as a sequence of::
+At a different level, a runner can take the form of an executable that
+follows the ``avocado-runner-$KIND`` naming pattern and conforms to a
+given interface/behavior, including accepting standardized command
+line arguments and producing standardized output.
+
+.. tip:: for a very basic example of the interface expected, refer to
+         ``selftests/functional/test_nrunner_interface.py`` on the
+         Avocado source code tree.
+
+Runner output
++++++++++++++
+
+A Runner should, if possible, produce status information on the
+progress of the execution of a Runnable.  While the Runner is
+executing what a Runnable describes, should it produce interesting
+information, the Runner should attempt to forward that along its
+generated status.
+
+For instance, using the ``exec`` Runner example, it's helpful to start
+producing status that the process has been created and it's running as
+soon as possible, even if no other output has been produced by the
+executable itself.  These can be as simple as a sequence of::
 
   {"status": "running"}
   {"status": "running"}
 
-When the process is finished, it can return::
+When the process is finished, the Runner may return::
 
   {"status": "finished", "returncode": 0, 'stdout': b'', 'stderr': b''}
 
-Note that, besides the status of ``finished``, and a return code which
-can be used to determine a success or failure status, it's not the
-runner's responsibility to determine test results.
+.. tip:: Besides the status of ``finished``, and a return code which
+         can be used to determine a success or failure status, a
+         Runner may not be obliged to determine the overall PASS/FAIL
+         outcome.  Whoever called the runner may be responsible to
+         determine its overall result, including a PASS/FAIL
+         judgement.
+
+Even though this level of information is expected to be generated by
+the Runner, whoever is calling a Runner, should be prepared to receive
+as little information as possible, and act accordingly.  That includes
+receiving no information at all.
```

Comment:
Ok, maybe this is answering my first question.

But I'm thinking out of loud: If I'm writing a new Runner for a specific test, should I also write the piece of code that will call my Runner? Or should I expect that Avocado will call it for me? (not sure it makes sense)

Reply:
You should expect Avocado to call it for you.  This is based on the requirement that the Runner input interface *MUST* be implemented in a compatible way.